### ISSUE TITLE
Android - first scan never works, works only after permissions are given.

### ISSUE THREAD

**Issue by Nixon-Joseph** [Redth/ZXing.Net.Mobile#741]
I'm running into this exact same issue as [this](https://github.com/Redth/ZXing.Net.Mobile/issues/297) on 2.4.1.

Here's my scanning page
```
public class BarcodeScanningPage : ContentPage
    {
        ZXingScannerView ZXingView { get; set; }
        ZXingDefaultOverlay Overlay { get; set; }

        public BarcodeScanningPage() : base()
        {
            Title = "Scanning...";

            ZXingView = new ZXingScannerView
            {
                HorizontalOptions = LayoutOptions.FillAndExpand,
                VerticalOptions = LayoutOptions.FillAndExpand,
                AutomationId = "zxingScannerView",
                Options = new ZXing.Mobile.MobileBarcodeScanningOptions()
                {
                    PossibleFormats = new List<ZXing.BarcodeFormat>() {
                        ZXing.BarcodeFormat.CODE_39, ZXing.BarcodeFormat.QR_CODE
                    }
                }
            };
            ZXingView.OnScanResult += (result) => Device.BeginInvokeOnMainThread(async () =>
            {
                // Stop analysis until we navigate away so we don't keep reading barcodes
                ZXingView.IsAnalyzing = false;

                await NavigationHelper.PushAsync(new SearchResultsPage(new ApiModels.SearchItemsParams()
                {
                    Keyword = result.Text,
                    ResultsPerPage = App.AppConfig.SearchResultsPerPage
                }, result.Text, true));

                //await NavigationHelper.PopAsync();
            });

            Overlay = new ZXingDefaultOverlay()
            {
                BottomText = "Scan Barcodes or QR Codes",
                ShowFlashButton = ZXingView.HasTorch,
                AutomationId = "zxingDefaultOverlay",
            };
            Overlay.FlashButtonClicked += (sender, e) =>
            {
                ZXingView.IsTorchOn = !ZXingView.IsTorchOn;
            };
            var grid = new Grid()
            {
                VerticalOptions = LayoutOptions.FillAndExpand,
                HorizontalOptions = LayoutOptions.FillAndExpand,
            };
            grid.Children.Add(ZXingView);
            grid.Children.Add(Overlay);

            // The root page of your application
            Content = grid;
        }

        protected override void OnAppearing()
        {
            base.OnAppearing();

            ZXingView.IsScanning = true;
        }

        protected override void OnDisappearing()
        {
            ZXingView.IsScanning = false;

            base.OnDisappearing();
        }
}
```
Any ideas how I might get around this? Thanks! Great plugin.

**Comment by rhedgpeth** [comment]
I am running into this issue as well across all Android devices with the latest version. Any possible workarounds?

**Comment by sbirdius** [comment]
Download version 2.3.1, it works fine to me. 
This is an old bug, but was fixed with  2.3.1 .However it seems to happen again in 2.4.1 ...

**Comment by Axemasta** [comment]
This issue is happening for me, the exact same code works on iOS perfectly!

**Comment by Axemasta** [comment]
Hi guys, this isn't a fix but instead a work around.

If you use [James Montemagno's permission library](https://github.com/jamesmontemagno/PermissionsPlugin) to check for permissions before you present your scanner, the android device will scan on first time load.

**Comment by derekwheeler-tylertech** [comment]
> If you use James Montemagno's permission library to check for permissions before you present your scanner, the android device will scan on first time load.

This work around didn't work for me.

**Comment by MKuckert** [comment]
Using 2.3.1 instead of 2.4.1 worked for me 🤷‍♂️

**Comment by pawanshakya** [comment]
Hi MKuckert - 
Do you have any sample code in regards which you can attach or any document on how to use this older version. 
I have used Zxing.Net.Mobile  &  Zxing.Net.Mobile.FOrms of both 2.3.1 version in Xamarin FOrm project but still facing similar issue.
Is there any specific setup configuration needs to be done ?

**Comment by matteopiccioni** [comment]
I solved using James Montemagno's permission library not into the 'scan class',
but on click button that go to scan class.
If user give permission then I go to scan class else I give error message

**Comment by TonyC5** [comment]
Reverting to ZXing.Net.Mobile 2.3.2 and ZXing.Net.Mobile.Forms 2.3.2 resolved the problem for me.

**Comment by zlannin** [comment]
I'm not familiar with Forms but Android needed the following code in my Activity. I'm running `2.4.1`.

    public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Permission[] grantResults)
    {
        base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        ZXing.Net.Mobile.Android.PermissionsHandler.OnRequestPermissionsResult(requestCode, permissions, grantResults);
    }

**Comment by Axemasta** [comment]
@zlannin The issue we are talking about is when using this in Xamarin Forms, you put this code into your MainActivity, and it never asks for your permissions and the camera never loads.

I have that code in my app and log when it triggers, it has never once triggered (other than the one time testing writing to my log file) over the 6 week development cycle of the app I was using the library in.

**Comment by reneposse** [comment]
I tried what @TonyC5 says. 
In MainActivity it works like this.
Works!!!!

`
protected override void OnCreate(Bundle bundle)
        {

            instance = this;

            TabLayoutResource = Resource.Layout.Tabbar;
            ToolbarResource = Resource.Layout.Toolbar;

            base.OnCreate(bundle);

            global::Xamarin.Forms.Forms.Init(this, bundle);
            global::ZXing.Net.Mobile.Forms.Android.Platform.Init();
          
            LoadApplication(new App());
        }
        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, Permission[] grantResults)
        {
            global::ZXing.Net.Mobile.Android.PermissionsHandler.OnRequestPermissionsResult(requestCode, permissions, grantResults);
            PermissionsImplementation.Current.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }`

**Comment by MaxxDelusional** [comment]
Overriding OnRequestPermissionsResult in my Activity did not work for me in 2.4.1

**Comment by Redth** [comment]
Thanks for reporting this issue!  Unforunately it took me way too long to respond 😭.  Sorry, I apologize!  Recently the source code for this project was completely refactored to modernize it.  Many PR's were included in this effort, and many bugs were hopefully fixed.  Please try out the latest 3.x series of NuGet packages (currently in prerelease).  To try and make the project more maintainable in my spare time going forward, I've decided to close all existing issues to start with a clean slate.  If you're still experiencing this issue on the newest version, please open a new issue with as much detail as possible.  Thank you for your patience and understanding!  Happy scanning!

**Comment by sunnyatticsoftware** [comment]
Upgrading to the `3.0.0-beta5` resolved the issue to me. Thanks

**Comment by bachvuviet** [comment]
I think if you check permission before Push the scan page you can prevent this problem :) Say the client have to give permission before the scan page appear

**Comment by dtekchandani** [comment]
@Redth By when 3.x(pre-release) version will be production ready?

**Comment by ghost** [comment]
> I think if you check permission before Push the scan page you can prevent this problem :) Say the client have to give permission before the scan page appear

I got the permission from user using Xamarin Essentials, before opening the scanner page. But still it is not working.
